src/components/API.js

```diff
@@ -0,0 +1,132 @@
+
+export default class API {
+  #baseURL
+  #headers
+  constructor(options){
+    this.#baseURL = options.baseURL;
+    this.#headers = options.headers; 
+  }
+
+  #handleResponse(response){
```

Comment:
Great that you've implemented a  method for processing server response, so the code that checks the server response and converts it from JSON  is reused